crates/apps/src/lib/node/ledger/shell/mod.rs

```diff
@@ -1108,6 +1110,20 @@ where
                     );
                     return response;
                 }
+                let batch_tx_hash = &tx.raw_header_hash();
```

Comment:
Replay protection is managed at the batch level, as in we check the `raw_header_hash` just like before. This hash contains the commitments of all the txs in the batch. This simplifies the logic and allows to (potenitally) reduce the number of signatures attached to the transaction. Txs in the batch are guaranteed to be different since the commitments are stored in a `HashSet`. Only drawback of this is that it's impossible to rewrap a single tx of the batch

Reply:
What do you mean by potentially reduce the number of signatures - as opposed to the wrapper tx fee payer signing over individual transactions within the batch in separate signatures?

Reply:
No I meant that by signing the entire header of the batch we need a signature per signer. If the signer of the inner txs we need only one signature for the entire batch. I still need to review if this is completely safe

Reply:
I think it's fine to bind the signatures for the inner transactions to the batch, if that's helpful.